src/services/anchor-service.ts

```diff
@@ -145,38 +145,53 @@ export class AnchorService {
+        merkleTree.getRoot().data.cid
+      )
 
-    // create proof on IPFS
-    logger.debug('Creating IPFS anchor proof')
-    const ipfsProofCid = await this._createIPFSProof(tx, merkleTree.getRoot().data.cid)
+      // create proof on IPFS
+      logger.debug('Creating IPFS anchor proof')
+      const ipfsProofCid = await this._createIPFSProof(tx, merkleTree.getRoot().data.cid)
 
-    // create anchor records on IPFS
-    logger.debug('Creating anchor commits')
-    const anchors = await this._createAnchorCommits(ipfsProofCid, merkleTree)
+      // create anchor records on IPFS
+      logger.debug('Creating anchor commits')
+      const anchors = await this._createAnchorCommits(ipfsProofCid, merkleTree)
 
-    // Update the database to record the successful anchors
-    logger.debug('Persisting results to local database')
-    const numAnchoredRequests = await this._persistAnchorResult(anchors, candidates)
+      // Update the database to record the successful anchors
+      logger.debug('Persisting results to local database')
+      const numAnchoredRequests = await this._persistAnchorResult(anchors, candidates)
 
-    logger.imp(`Service successfully anchored ${anchors.length} CIDs.`)
-    return {
-      acceptedRequestsCount: groupedRequests.acceptedRequests.length,
-      alreadyAnchoredRequestsCount: groupedRequests.alreadyAnchoredRequests.length,
-      anchoredRequestsCount: numAnchoredRequests,
-      conflictingRequestCount: groupedRequests.conflictingRequests.length,
-      failedRequestsCount: groupedRequests.failedRequests.length,
-      failedToPublishAnchorCommitCount: merkleTree.getLeaves().length - anchors.length,
-      unprocessedRequestCount: groupedRequests.unprocessedRequests.length,
-      candidateCount: candidates.length,
-      anchorCount: anchors.length,
+      logger.imp(`Service successfully anchored ${anchors.length} CIDs.`)
+      return {
+        acceptedRequestsCount: groupedRequests.acceptedRequests.length,
+        alreadyAnchoredRequestsCount: groupedRequests.alreadyAnchoredRequests.length,
+        anchoredRequestsCount: numAnchoredRequests,
+        conflictingRequestCount: groupedRequests.conflictingRequests.length,
+        failedRequestsCount: groupedRequests.failedRequests.length,
+        failedToPublishAnchorCommitCount: merkleTree.getLeaves().length - anchors.length,
+        unprocessedRequestCount: groupedRequests.unprocessedRequests.length,
+        candidateCount: candidates.length,
+        anchorCount: anchors.length,
+      }
+    } catch (err) {
+      logger.imp(
+        `Updating requests to pending to be retried in the next batch because an error occured while creating the anchors: ${err}`
+      )
+
+      // TODO(NET-1623): Add alert that something went wrong and we are retrying
+
+      await this.requestRepository.updateRequests(
+        { status: RequestStatus.PENDING },
+        groupedRequests.acceptedRequests
+      )
+
+      throw err
```

Comment:
what happens to the overall anchor batch when this throws?

Reply:
we throw on a lot of cases (ex. ipfs error, insufficient funds, eth provider errors) I'm assuming our current alert system depends on this. The overall anchor batch gets lost forever, nothing persists it. The requests fail in w.e status they are left off in (PROCESSING, FAILED or COMPLETED)